### PR TITLE
[Refactor] modified to clang-format & checked with cpplint (google code style)

### PR DESCRIPTION
**Pull request by JieRen98**

### CODE REVIEW

File: src/resource/handle_manager.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+* Copyright (c) 2021 Megvii Inc. All rights reserved.
+*
+**/
+
+#include "resource/handle_manager.h"
+#include <cassert>
+#include "resource/memory_pool.h"
+
+namespace MegBA {
+void HandleManager::createNcclComm() {
```
Comment (bitterengsci) [inline src/resource/handle_manager.cpp:13]: Nccl -> NCCL?

File: src/resource/handle_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+const std::vector<ncclComm_t> &HandleManager::getNcclComm() { return comms; }
+
+void HandleManager::destroyNcclComm() {
+  for (auto comm : comms) {
+    ncclCommDestroy(comm);
+  }
+}
+
+void HandleManager::createCublasHandle() {
```
Comment (bitterengsci) [inline src/resource/handle_manager.cpp:30]: Cublas -> CUBLAS?

File: src/resource/handle_manager.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+void HandleManager::destroyCublasHandle() {
+  std::unique_lock<std::mutex> lock{mutex};
+  for (int i = 0; i < cublasHandle.size(); ++i) {
+    cudaSetDevice(i);
+    cublasDestroy_v2(cublasHandle[i]);
+  }
+  cublasHandle.clear();
+}
+
+void HandleManager::createCusparseHandle() {
```
Comment (bitterengsci) [inline src/resource/handle_manager.cpp:55]: Cusparse -> CUSPARSE?

File: include/resource/handle_manager.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  static std::mutex mutex;
+
+ public:
+  static void createNcclComm();
+
+  static const std::vector<ncclComm_t> &getNcclComm();
+
+  static void destroyNcclComm();
+
+  static void createCublasHandle();
```
Comment (bitterengsci) [inline include/resource/handle_manager.h:30]: Cublas -> CUBLAS?

File: include/resource/handle_manager.h [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  static const std::vector<cublasHandle_t> &getCublasHandle();
+
+  static void destroyCublasHandle();
+
+  static void createCusparseHandle();
```
Comment (bitterengsci) [inline include/resource/handle_manager.h:36]: Cusparse -> CUSPARSE?

File: include/operator/jet_vector_op-inl.h [hunk 6]
```diff
@@ -72,41 +71,39 @@ namespace MegBA {
...
-                    function::sqrt_JetVector_CUDA(f, out);
-                    break;
-                }
-            }  // switch device_
-            return out;
-        }
+// sqrt(a + h) ~= sqrt(a) + h / (2 sqrt(a))
+template <typename T> inline JetVector<T> sqrt(JetVector<T> f) {
+  JetVector<T> out;
+  out.Init_as(f);
```
Comment (bitterengsci) [inline include/operator/jet_vector_op-inl.h:77]: Init_as 这个是不是要改一下？

**Review by bitterengsci — COMMENTED** [review]
Please check my comments

File: include/operator/jet_vector_op-inl.h (hunk 6 above)
Comment (Irwin-Liu) [reply include/operator/jet_vector_op-inl.h:77]: initAs?

File: include/resource/handle_manager.h (hunk 5 above)
Comment (Irwin-Liu) [reply include/resource/handle_manager.h:36]: i think CuBLAS or CUBLAS

File: include/resource/memory_pool.h [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+  static void allocateNormal(void **ptr, size_t size, int rank = 0);
+
+  static void deallocateNormal(void *ptr, int rank = 0);
+
+  static int getWorldSize() { return _worldSize; }
+
+  static void redistribute();
+
+  static std::size_t getElmNum(int rank) {
```
[the remark this answers is not in the document]
Comment (Irwin-Liu) [reply include/resource/memory_pool.h:43]: agree

File: include/resource/handle_manager.h [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace MegBA {
+class HandleManager {
+  static std::vector<ncclComm_t> comms;
+  static std::vector<cublasHandle_t> cublasHandle;
+  static std::vector<cusparseHandle_t> cusparseHandle;
+  static std::mutex mutex;
+
+ public:
+  static void createNcclComm();
```
[the remark this answers is not in the document]
Comment (Irwin-Liu) [reply include/resource/handle_manager.h:24]: agree

File: include/operator/jet_vector_math_impl.cuh [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace MegBA {
+namespace math {
+namespace impl {
+template <typename T>
+void vectorAddVectorCUDA(const JetVector<T> &f, const JetVector<T> &g,
+                         JetVector<T> *out);
+
+template <typename T>
+void vectorMinusVectorCUDA(const JetVector<T> &f, const JetVector<T> &g,
```
[the remark this answers is not in the document]
Comment (Irwin-Liu) [reply include/operator/jet_vector_math_impl.cuh:19]: agree

File: include/wrapper.hpp [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+**/
+
+#pragma once
+#include <cublas_v2.h>
+#include <cusparse_v2.h>
+#include <nccl.h>
+#include <utility>
+#include <tuple>
+
+__device__ float rsqrtf(float a);
```
[the remark this answers is not in the document]
Comment (JieRen98) [reply include/wrapper.hpp:15]: 这里不是重载，rsqrtf 是已有的 device function，需要声明，不然会有链接错误

File: include/operator/jet_vector-inl.h [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+* MegBA is Licensed under the Apache License, Version 2.0 (the "License")
+*
+* Copyright (c) 2021 Megvii Inc. All rights reserved.
+*
+**/
+
+#pragma once
+
+namespace MegBA {
+namespace CHK {
```
Comment (bitterengsci) [inline include/operator/jet_vector-inl.h:11]: CHK meaning?

File: include/operator/jet_vector-inl.h [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+template <typename T>
+inline bool Device_Same(const JetVector<T> &f, const JetVector<T> &g) {
```
Comment (bitterengsci) [inline include/operator/jet_vector-inl.h:13]: Device_Same -> DeviceSame?

File: include/operator/jet_vector-inl.h [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+inline void deviceThrow(const JetVector<T> &f, const JetVector<T> &g) {
+
+  if (!Device_Same(f, g))
+    throw std::runtime_error("Different _device for item #1 is on " +
+                             std::to_string(f.getDevice()) + " item #2 is on " +
+                             std::to_string(g.getDevice()));
+}
+
+template <typename T>
+inline bool Shape_Same(const JetVector<T> &f, const JetVector<T> &g) {
```
Comment (bitterengsci) [inline include/operator/jet_vector-inl.h:27]: Shape_Same -> shapeSame?

File: src/edge/base_edge.cpp [hunk 14]
```diff
@@ -0,0 +1,391 @@
...
+  absolutePositionCamera = nullptr;
+  absolutePositionPoint = nullptr;
+  connectionNumPoint = nullptr;
+}
+
+template <typename T>
+EdgeVector<T>::EdgeVector(const ProblemOption &option,
+                          const std::vector<SchurHessianEntrance<T>> &schurHessianEntrance)
+    : _option{option}, schurHessianEntrance{schurHessianEntrance},
+      schurCsrRowPtr(option.deviceUsed.size()) {
```
Comment (bitterengsci) [inline src/edge/base_edge.cpp:90]: Csr -> CSR?

File: src/edge/base_edge.cpp [hunk 15]
```diff
@@ -0,0 +1,391 @@
...
+    edgeKind = TWO_CAMERA;
+  else
+    edgeKind = MULTI;
+}
+
+template <typename T> void EdgeVector<T>::SchurEquationContainer::clear() {
+  switch (_device) {
+  case Device::CPU: {
+    for (int i = 0; i < 2; ++i)
+      free(csrRowPtr[i]);
```
Comment (bitterengsci) [inline src/edge/base_edge.cpp:37]: csr -> CSR?

File: src/edge/base_edge.cpp [hunk 16]
```diff
@@ -0,0 +1,391 @@
...
+  schurEquationContainer.reserve(option.deviceUsed.size());
+  for (int i = 0; i < option.deviceUsed.size(); ++i) {
+    schurEquationContainer.emplace_back(option.device);
+    schurPositionAndRelationContainer.emplace_back(option.device);
+  }
+}
+
+template <typename T> void EdgeVector<T>::rollback() {
+  if (_option.useSchur) {
+    schurDaPtrs.swap(schurDaPtrsOld);
```
Comment (bitterengsci) [inline src/edge/base_edge.cpp:100]: what is Da? Vague

File: include/operator/jet_vector-inl.h (hunk 11 above)
Comment (Irwin-Liu) [reply include/operator/jet_vector-inl.h:11]: cholesky? suggest use full word

File: include/operator/jet_vector-inl.h (hunk 12 above)
Comment (Irwin-Liu) [reply include/operator/jet_vector-inl.h]: agree

File: include/operator/jet_vector-inl.h (hunk 13 above)
Comment (Irwin-Liu) [reply include/operator/jet_vector-inl.h]: agree

File: src/edge/base_edge.cpp (hunk 14 above)
Comment (Irwin-Liu) [reply src/edge/base_edge.cpp:90]: agree

File: src/edge/base_edge.cpp (hunk 16 above)
Comment (Irwin-Liu) [reply src/edge/base_edge.cpp:100]: use full word?

**Pull request merged**